eessi/testsuite/tests/apps/walberla/walberla.py

```diff
@@ -0,0 +1,154 @@
+    file from the precompiled walberla tutorials binary. These are run as pre-run commands and the main test with the
+    executable `05_BackwardFacingStep` is measured for performance.
+    """
+#    ldc_64M = fixture(EESSI_OPENFOAM_base, scope='partition')
+    executable = '05_BackwardFacingStep'
+    executable_opts = ['05_BackwardFacingStep.prm']
+    time_limit = '120m'
+    readonly_files = ['']
+    device_type = parameter([DEVICE_TYPES.CPU])
+    module_name = parameter(find_modules('waLBerla'))
+    valid_systems = ['*']
+    scale = parameter(filter_scales_1M())
+
+    @run_after('init')
+    def set_compute_unit(self):
+        """
+        Set the compute unit to which tasks will be assigned:
+        one task per CPU core for CPU runs.
+        """
+        if self.device_type == DEVICE_TYPES.CPU:
+            self.compute_unit = COMPUTE_UNITS.CPU
+        else:
+            msg = f"No mapping of device type {self.device_type} to a COMPUTE_UNITS was specified in this test"
+            raise NotImplementedError(msg)
+
+    def required_mem_per_node(self):
+        return self.num_tasks_per_node * 1700
+
+    @run_after('setup')
+    def check_launcher_options(self):
+        # We had to get the launcher command and prepend this to the prerun steps (func prepare_environment) because:
+        # 1. A typical OpenFOAM job would contain multiple mpirun steps working on the same stage directory.
+        # 2. We had trouble using ReFrame fixtures to separate these over multiple jobs, because we couldn't get it to
+        #    work together with the mixin class.
+        if (self.job.launcher.command(self.job)[0] == 'mpirun'):
+            self.launcher_command = self.job.launcher.command(self.job)
+            self.launcher_command[-1] = str(self.num_tasks_per_node * self.num_nodes)
+        elif (self.job.launcher.command(self.job)[0] == 'srun'):
+            self.launcher_command = self.job.launcher.command(self.job)
+        else:
+            self.skip(msg="The chosen launcher for this test is different from mpirun or srun which means that the"
+                      "test will definitely fail, therefore skipping this test.")
+
+    @run_after('setup')
+    def check_core_count(self):
+        # The walberla backward facing step test case can run on a max of 384 cores and a minimum of 8 cores to run
+        # within reasonable time.
+        if self.num_tasks > 384:
+            self.skip(msg="The maximum number of cores that the strong scaling can hold for this test is 384. Launch on"
+                      "a scale with a smaller core count.")
+        elif self.num_tasks < 8:
+            self.skip(msg="The minimum number of MPI tasks required for this test is 8 to run within reasonable time."
+                      "Launch of scale with larger core count")
+
+    @run_before('run')
+    def prepare_environment(self):
+        # fullpath = os.path.join(self.ldc_64M.stagedir, 'fixedTol')
+        self.prerun_cmds = [
```

Comment:
Please add inline comments here to document what the different `sed` commands do. Ideally, give an example of a before - after situation. This will be very helpful if someone ever needs to debug this, as from the `sed` statements alone it is very hard to determine what was _supposed_ to happen.